fedimint-client/src/meta.rs

```diff
@@ -0,0 +1,244 @@
+            )
+        }))
+        .await;
+        // intially write all once
+        let mut dbtx = client.db().begin_transaction().await;
+        for (source, meta) in metas {
+            if let Ok(meta) = meta.and_then(|x| x) {
+                Self::write_source_meta(&mut dbtx, &**source, &meta).await;
+            }
+        }
+        dbtx.commit_tx().await;
+
+        // now keep updating all
+        let values = join_all(
+            self.sources
+                .iter()
+                .map(|source| self.update_source_continuously(client, &**source)),
+        )
+        .await;
+        // to get ! type
+        values[0]
+    }
+
+    async fn update_source_continuously(&self, client: &Client, source: &dyn MetaSource) -> ! {
+        loop {
+            MetaSource::wait_for_update(source).await;
+            if let Ok(meta) = retry(
+                "update_source",
+                // very slow retry
+                FibonacciBackoff::default()
+                    .with_min_delay(Duration::from_secs(10))
+                    .with_max_delay(Duration::from_secs(10 * 60))
+                    .with_max_times(usize::MAX),
+                || source.fetch(client),
+            )
+            .await
+            {
+                let mut dbtx = client.db().begin_transaction().await;
+                Self::write_source_meta(&mut dbtx, source, &meta).await;
+                let _ = dbtx.commit_tx_result();
+            }
+        }
+    }
+
+    async fn write_source_meta(
+        dbtx: &mut DatabaseTransaction<'_, Committable>,
+        source: &dyn MetaSource,
+        meta: &BTreeMap<String, String>,
+    ) {
+        let priority = source.priority();
+        for (key, value) in meta {
+            dbtx.insert_entry(
+                &MetaFieldKey {
+                    name: MetaFieldName(key.clone()),
+                    priority,
+                },
+                &value,
+            );
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+#[non_exhaustive]
+pub struct LegacyClientConfigMetaSource {}
+
+#[apply(async_trait_maybe_send!)]
+impl MetaSource for LegacyClientConfigMetaSource {
+    fn priority(&self) -> u64 {
+        0
+    }
+
+    async fn wait_for_update(&self) {
+        pending().await
+    }
+
+    async fn fetch(&self, client: &Client) -> anyhow::Result<BTreeMap<String, String>> {
+        Ok(client.config().global.meta.clone())
+    }
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct LegacyMetaOverrideUrl {
+    client: reqwest::Client,
+}
+
+#[apply(async_trait_maybe_send!)]
+impl MetaSource for LegacyMetaOverrideUrl {
+    fn priority(&self) -> u64 {
+        1
+    }
+
+    async fn wait_for_update(&self) {
+        // update every 10 minutes
+        fedimint_core::task::sleep(Duration::from_secs(10 * 60)).await
+    }
+
+    async fn fetch(&self, client: &Client) -> anyhow::Result<BTreeMap<String, String>> {
+        let Some(url) = client.config().meta::<String>("TODO")? else {
```

Comment:
Let's make the field name a parameter stored in the struct. The default constructor would use the standardized `meta_override_url`, but for compatibility reasons the pre-standardization variants could also be used.